### PR TITLE
Improve addition assignment example

### PR DESCRIPTION
**Pull request by SMYusupov**
#### Summary
Fix expected result, make code more readable

#### Motivation
Make code more readable and don't be misled

This PR…
-->
- [ ] Make expected result correct
- [ ] Rewrite code to make it more readable

### CODE REVIEW

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks for your PR, @P1GAS !

I agree that we could clarify this example and make it more "real".

But this change is not quite right. I think the existing page is expecting that each assignment is being made on the original value of the variable.

For example, here:

```js
// Boolean + Number -> addition
baz += 1 // 2

// Boolean + Boolean -> addition
baz += false // 2
```

...in the second addition assignment, it says `baz` is boolean, but after the first one it is a number.

Similarly, your change here:

```js
baz += false // 2
```

...seems to be because you previously incremented `baz`, but that's inconsistent with the later one:

```js
let foo = 'foo'

// String + Boolean -> concatenation
foo += false // "foofalse"

// String + String -> concatenation
foo += 'bar' // "foobar"
```

...why isn't the last value here "foofalsebar"?

I think it would be best to make it very clear that these are all self-contained examples, perhaps by having separate code blocks for them? For example:

```js
let bar = 5;

// Number + Number -> addition
bar += 2; // 7
```

```js
let bar = 5;

// Number + String -> concatenation
bar += 'foo'; // "5foo"
```
...etc

**Comment by P1GAS** [comment]
Hello @wbamberg! I have changed the code, could you check it again?

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thank you for the update @P1GAS . But this isn't the change I asked for. I suggested making each assignment a separate self-contained example. Is there a reason you don't think this is a good idea?

**Comment by P1GAS** [comment]
I think that's a good idea. I have another view similar to yours. What if we create 3 blocks with every variable

**Comment by wbamberg** [comment]
> @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> 

Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.

**Comment by teoli2003** [comment]
> > @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> 
> Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.

These must be let as the operator `+=` is used. But I agree for the semi-colon.

**Comment by wbamberg** [comment]
> > > @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> > 
> > 
> > Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.
> 
> These must be let as the operator `+=` is used. But I agree for the semi-colon.

Doh, sorry! Yes of course.

File: files/en-us/web/javascript/reference/operators/addition_assignment/index.md [hunk 1]
```diff
@@ -21,36 +21,41 @@ possible.
 ## Syntax
 
 ```js
-x += y // x = x + y
+x += y; // x = x + y
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/addition_assignment/index.md:24]:
```suggestion
x += y // x = x + y
```

**Review by wbamberg — APPROVED** [review]
👍 looks good to me, thank you @P1GAS !

**Pull request merged**